src/integer64.c

```diff
@@ -156,7 +156,17 @@ SEXP as_integer64_character(SEXP x_, SEXP ret_){
   char * endpointer;
   for(i=0; i<n; i++){
     str = CHAR(STRING_ELT(x_, i)); endpointer = (char *)str; // thanks to Murray Stokely 28.1.2012
-    ret[i] = strtoll(str, &endpointer, 10);
+    int base = 10; // default
+    int sign = 1;  // default 
+    if (str[0]=='-'){
+      sign = -1;
+      str++;
```

Comment:
`str` is `const` above, this will violate that, won't it?

Reply:
If `str` is a pointer-to-const, this does not violate the const-ness. We are not writing through the pointer to modify `*str`; we are only incrementing our copy of `str`, which does not modify the value in the caller.

Reply:
oh right, basically it's `(const str) *`, not `const (str *)`